app/modules/asset_groups/models.py

```diff
@@ -786,226 +816,298 @@ def build_detection_request(self, job_uuid, model):
             _external=True,
         )
 
-        ia_config_reader = IaConfig(current_app.config.get('CONFIG_MODEL'))
-        detector_config = ia_config_reader.get_named_detector_config(model)
-
-        assert 'start_detect' in detector_config
-
         model_config = {
-            'endpoint': detector_config['start_detect'],
             'jobid': str(job_uuid),
             'callback_url': f'houston+{callback_url}',
             'callback_detailed': True,
         }
+
+        ia_config_reader = IaConfig(current_app.config.get('CONFIG_MODEL'))
+        detector_config = ia_config_reader.get_named_detector_config(model)
         model_config.update(detector_config)
 
-        asset_urls = []
+        assets = []
         if 'updatedAssets' in self.sighting_config:
-            asset_urls = [
-                url_for('api.assets_asset_src_raw_by_id', asset_guid=guid, _external=True)
-                for guid in self.sighting_config['updatedAssets']
-            ]
+            for guid in self.sighting_config['updatedAssets']:
+                asset = Asset.query.get(guid)
+                if not asset:
+                    logging.warning(f'Asset ref {guid}, cannot find asset, skipping')
+                    continue
+                assets.append(asset)
         else:
             for filename in self.sighting_config.get('assetReferences'):
                 asset = self.asset_group.get_asset_for_file(filename)
                 if not asset:
                     logging.warning(f'Asset ref {filename}, cannot find asset, skipping')
                     continue
+                assets.append(asset)
 
-                asset_url = url_for(
-                    'api.assets_asset_src_raw_by_id',
-                    asset_guid=str(asset.guid),
-                    _external=True,
+        asset_sage_data = []
+
+        if preload:
```

Comment:
If `preload=True`, we preemptively send the Assets to Sage by uploading its file bytes to Sage.  This returns a UUID and we store that on the Asset as its content_guid.  If the content_guid is specified on an asset, we ensure that it exists by checking with Sage.  If it doesn't, `asset.sync_with_sage(ensure=True)` will reset and force push the Asset to Sage.